packages/Badge/index.tsx

```diff
@@ -1,12 +1,62 @@
 import React from 'react'
-import { Tag, TagOptions } from '@welcome-ui/tag'
 import { CreateWuiProps, forwardRef } from '@welcome-ui/system'
 
-export type BadgeOptions = Omit<TagOptions, 'size'>
+import * as S from './styles'
+
+export type BadgeOptions = {
+  disabled?: boolean
+  shape?: 'circle' | 'square'
+  size?: 'sm' | 'md'
+  variant?: 'default' | 'primary'
+  // if a number is higher than 99, we replace this number by 99+
+  withNumberAbbreviation?: boolean
+  children: string | number
+}
 export type BadgeProps = CreateWuiProps<'div', BadgeOptions>
 
-export const Badge = forwardRef<'div', BadgeProps>((props, ref) => (
-  <Tag ref={ref} size="sm" {...props} />
-))
+export const Badge = forwardRef<'div', BadgeProps>(
+  (
+    {
+      children,
+      dataTestId,
+      disabled,
+      shape = 'circle',
+      size = 'md',
+      variant = 'default',
+      withNumberAbbreviation,
+      ...rest
+    },
+    ref
+  ) => {
+    let text
+    const isNumber = Number.isInteger(children)
+    const textLength = children.toString().length
```

Comment:
Ah that's why we want only strings or number

Reply:
yes :)